app/models/notify_user/scheduler.rb

```diff
@@ -0,0 +1,46 @@
+module NotifyUser
+  class Scheduler
+    def self.schedule(notification)
+      new(notification).schedule
+    end
+
+    def initialize(notification)
+      @notification = notification
+    end
+
+    def schedule
+      notification.class.channels.each do |channel, options|
+        aggregator = Aggregator.new(notification, options[:aggregate_per])
+        return if aggregator.has_pending_deliveries?
+
+        # Create the delivery:
+        delivery = Delivery.create!({
+          notification: notification,
+          deliver_in: delay_time(aggregator),
+          channel: channel.to_s
+        })
+      end
+    end
+
+    private
+
+    attr_reader :notification
+
+    # Find the seconds to delay delivery by:
+    def delay_time(aggregator)
+      last_send_time = aggregator.last_send_time
+      delay_time = aggregator.delay_time_in_seconds
+
+      validate_delay_time(last_send_time, delay_time)
```

Comment:
To me, `validate_` sounds like it has side-effects rather than a return value. Maybe `validated_delay_time` is a better name?